rust/core-lib/src/tabs.rs

```diff
@@ -371,9 +372,13 @@ impl<W: Write + Send + 'static> Documents<W> {
             let buffer_id = self.next_buffer_id();
             self.new_empty_view(rpc_peer, &view_id, buffer_id);
         }
+
+        // schedule plugin notification for after the view has received an identifer
+        let view_id2 = view_id.to_owned();
         let init_info = self.buffers.lock().editor_for_view(&view_id)
             .unwrap().plugin_init_info();
-        self.plugins.document_new(&view_id, init_info);
+        self.plugins.run_on_idle(move |p| p.document_new(&view_id2, &init_info));
```

Comment:
It does look better now but here's a different idea:

Accidentally sending notifications while responding to a RPC call with a reply sounds like it will be a recurring problem. Instead of having to know which things will send notifications and carefully defer them (and what if you need the result of the notifying thing to craft your reply?!), you should be able to send notifications whenever and have them be delivered correctly. What I'm thinking is that when you tell the event loop to send a notification, it knows if it is already in the middle of a procedure call and if so pushes the notifications onto a queue, which it then sends off when the call is done.

I don't like the current `run_on_idle` method for a number of reasons:

1. Queueing `run_on_idle` tasks sounds like the event loops job, not the plugin manager's.
1. "idle" doesn't really mean idle, there might be a thousand events queued up waiting to execute and you'll still run it on the next spin right? If not, does that mean document creation has unbounded latency? I feel that an idle mechanism should only be used for things you actually want to run on idle, like garbage collection, not things you just want to run "later because otherwise it'll mess things up, but not too late please!".
1. I can't tell if there's a race condition here where the frontend immediately sends a command to a plugin after creating a new view, but the plugin manager doesn't know the document has been created yet because the new event arrived before the idle call.
1. It feels like it might end up like Cocoa/IOS where if something doesn't work you shrug and wrap it in a timer that runs after 0.1ms and it fixes it without you even knowing why.